Docker/validate.py

```diff
@@ -0,0 +1,158 @@
+                        help="synapse entity type downloaded")
+    parser.add_argument("-s", "--submission_file", help="Submission file")
+    parser.add_argument("-c", "--condition", required=True, nargs='+',
+                        help="Experiment condition")
+    parser.add_argument("-p", "--proportion", required=True, nargs='+',
+                        help="Downsampling proportion")
+    parser.add_argument("-x", "--file_prefix", required=True,
+                        help="Prefix of filename")
+    parser.add_argument("-q", "--question", required=True,
+                        help="Challenge question")
+    return parser.parse_args()
+
+
+def _filter_files(members, type="tar"):
+    """Filter out non-csv files in zip file."""
+    if type == "tar":
+        new_members = filter(
+            lambda member: member.name.endswith(".csv"), members)
+    else:
+        new_members = filter(lambda member: member.endswith(".csv"), members)
+    new_members = list(new_members)
+    return new_members
+
+
+def _decompress_file(f):
+    """Untar or unzip file."""
+    names = []
+    # decompress zip file
+    if zipfile.is_zipfile(f):
+        with zipfile.ZipFile(f) as zip_ref:
+            members = zip_ref.namelist()
+            members = _filter_files(members, type="zip")
+            if members:
+                for member in members:
+                    member_name = os.path.basename(member)
+                    with zip_ref.open(member) as source, open(member_name, 'wb') as target:
+                        # copy it directly to skip the folder names
+                        shutil.copyfileobj(source, target)
+                    names.append(member_name)
+    # decompress tar file
+    elif tarfile.is_tarfile(f):
+        with tarfile.open(f) as tar_ref:
+            members = tar_ref.getmembers()
+            members = _filter_files(members)
+            if members:
+                for member in members:
+                    # skip the folder names
+                    member.name = os.path.basename(member.name)
+                    tar_ref.extract(member)
+                    names.append(member.name)
+    return names
+
+
+def _validate_scRNA(ds_files, pred_files):
+    """validate on scRNA-seq data"""
+    invalid_reasons = []
+    # validate each prediction file
+    for index, pred_f in enumerate(pred_files):
+        # read ds file
+        ds_df = pd.read_csv(ds_files[index], index_col=0)
+        # read corresponding pred file
+        pred_df = pd.read_csv(pred_f, index_col=0)
+        # check if all genes exist in predictions
+        cp1 = set(list(pred_df.index)).issubset(list(ds_df.index))
+        # check if all cells exist in predictions
+        cp2 = set(list(pred_df.columns.values)).issubset(
+            list(ds_df.columns.values))
+        if not (cp1 and cp2):
+            invalid_reasons.append(
+                pred_f + ": Do not contain all genes or cells")
```

Comment:
nbd here, but I recommend using f-strings instead of concatenating with the `+` operator.  I find f-strings easier to read, and they're a bit more convenient to use, in that it doesn't require all inputs to be strings (unlike `+`)